waspc/src/Wasp/Generator/WebAppGenerator/RouterGenerator.hs

```diff
@@ -25,15 +27,19 @@ import qualified Wasp.Generator.WebAppGenerator.Common as C
 data RouterTemplateData = RouterTemplateData
   { _routes :: ![RouteTemplateData],
     _pagesToImport :: ![PageTemplateData],
-    _isAuthEnabled :: Bool
+    _isAuthEnabled :: Bool,
+    _isExternalAuthEnabled :: Bool,
+    _isGoogleAuthEnabled :: Bool
```

Comment:
Seems like these variables are coupled in a way that isn't clear from their type. For example, the following configuration is invalid and we can only know this by understanding the hidden relationship between the variables.
```haskell
_isExternalAuthEnabled: false
_isGoogleAuthEnabled: true
```
The same applies the other way around, although this is due to Google being the only external auth method we currently support.

I'm guessing, since this is a template, there is no obvious way to improve the code?

Reply:
Yeah that is a good point, the dependency is not clear from the names/types once you get into the template. Since we only want to import something if external auth is enabled, meaning Google (or one of the others in future), I'm not sure how best we can handle:

```js
{=# isExternalAuthEnabled =}
import OAuthCodeExchange from "./auth/pages/OAuthCodeExchange"
{=/ isExternalAuthEnabled =}
```

Reply:
At least further down in the file, this outer check makes the relationship clearer (but is not strictly required):

```js
      {=# isExternalAuthEnabled =}

      {=# isGoogleAuthEnabled =}
      <Route exact path="/auth/redirect/google">
        <OAuthCodeExchange pathToApiServerRouteHandlingOauthRedirect="/auth/external/google/validateCode" />
      </Route>
      {=/ isGoogleAuthEnabled =}

      {=/ isExternalAuthEnabled =}
```

Reply:
Yeah, I see no better way so I'm resolving :smile: